### PR TITLE
Concordance performance improvements

### PR DESCRIPTION
**Pull request by h-croser**
Performance improvements for concordance.py that target unnecessary components within the pipeline for processing text through the SpaCy model.
Additionally, the Jupyter Notebook widget for keyword search no longer updates on every character entry, but instead waits for user to submit to perform search, which reduces the number of searches performed.

### CODE REVIEW

File: src/atap_widgets/concordance.py [hunk 1]
```diff
@@ -127,18 +128,20 @@ def prepare_text_df(
...
-        output["text_id"] = pd.Series(range(output.shape[0]), dtype=pd.Int64Dtype)
+        output["text_id"] = pd.Series(range(output.shape[0]), dtype=pd.Int64Dtype())
         id_column = "text_id"
     output = output.rename(columns={text_column: "text", id_column: "text_id"})
     output = output.set_index("text_id", drop=False)
 
     if isinstance(language_model, str):
         language_model = spacy.load(language_model)
-    output["spacy_doc"] = output["text"].map(language_model)  # Doc for each line
+    output["spacy_doc"] = [
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:142]: This looks like a good way to get some easy performance improvements. Looks like `pipe()` has an `n_process` argument, have you checked whether we get additional improvements using multiple processors?

File: src/atap_widgets/concordance.py [hunk 2]
```diff
@@ -108,7 +109,7 @@ def prepare_text_df(
     df: pd.DataFrame,
     text_column: str = "text",
     id_column: str = None,
-    language_model: Union[str, spacy.language.Language] = "en_core_web_sm",
+    language_model: Union[str, spacy.language.Language] = English(),
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:112]: I hadn't seen this way of setting up a spacy model, and don't fully understand what it's doing relative to using `en_core_web_sm` but it does look pretty readable, so I'm happy to go with this as the default.

Can you check what components the `English` model uses and describe them in a comment or the docstring, so people know what components are necessary and which can be disabled?

File: src/atap_widgets/concordance.py [hunk 3]
```diff
@@ -340,7 +341,9 @@ def display_results(page: int, **kwargs):
 
             display(ipywidgets.HTML(html))
 
-        keyword_input = ipywidgets.Text(description="Keyword(s):")
+        keyword_input = ipywidgets.Text(
```
Comment (marius-mather) [inline src/atap_widgets/concordance.py:344]: `continuous_update` obviously has performance impacts but I think it is nice to have in an interactive tool, can we have it as an argument to `show()` so you can turn it on or off?

**Review by marius-mather — APPROVED** [review]
Looks good to me! 

A couple of comments/suggestions but otherwise this looks sensible as a way to improve performance. If you're able to do a quick check of whether `n_process` further improves performance on a large corpus, that would be great, but it's not a hard requirement if you're already getting acceptable performance.

File: src/atap_widgets/concordance.py (hunk 1 above)
Comment (h-croser) [reply src/atap_widgets/concordance.py:142]: When I increase this argument the benchmark I'm using takes twice as long to run

File: src/atap_widgets/concordance.py (hunk 2 above)
Comment (h-croser) [reply src/atap_widgets/concordance.py:112]: So calling `English()` is equivalent to `spacy.blank("en")`, so there are actually no components loaded. I'll add this detail in the comments

File: src/atap_widgets/concordance.py (hunk 3 above)
Comment (h-croser) [reply src/atap_widgets/concordance.py:344]: I could include a toggle, but I think for this specific input field it's fine to leave `continuous_update` False, as the user just has to press enter or deselect to start the search process. I intend to meet with a user next week to get their input about this change. Should I leave this out of this PR and include it in a future change?

Comment (marius-mather) [reply src/atap_widgets/concordance.py:344]: Yeah ask the user whether they'd want it or not.

**Pull request merged**